EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/GeneratePanel.java

```diff
@@ -242,4 +308,128 @@ private void addSubmitButton() {
+                        return selectedEventType.getTypeName() + "-values.json";
+                    }
+                };
+                GeneratePanel.this.add(jsonDownload);
+                jsonDownload.initiate(target);
+
+                GeneratePanel.this.page.getFeedbackPanel().success("Json created.");
+                target.add(GeneratePanel.this.page.getFeedbackPanel());
+            }
+        };
+        layoutForm.add(exportButton);
+    }
+
+    /**
+     * Add a upload field to upload a file.
+      */
+    private void addImportField() {
+        uploadField = new FileUploadField("importValuesUpload");
+        this.importForm.add(uploadField);
+    }
+
+    /**
+     * Add a button to import values from a file.
+     */
+    private void addImportSubmitButton() {
+        AjaxButton button = new AjaxButton("importValuesButton") {
+            @Override
+            public void onSubmit(AjaxRequestTarget target, Form importForm) {
+                final FileUpload uploadedFile = uploadField.getFileUpload();
+                if (uploadedFile == null) {
+                    GeneratePanel.this.page.getFeedbackPanel().error("File not found.");
+                    target.add(GeneratePanel.this.page.getFeedbackPanel());
+                    return;
+                }
+                // make sure provided file is json
+                final String fileName = uploadedFile.getClientFileName();
+                String fileFormat = fileName.substring(fileName.lastIndexOf('.') + 1);
+                if (!"json".equals(fileFormat)) {
+                    GeneratePanel.this.page.getFeedbackPanel().error("Please provide a json file.");
+                    target.add(GeneratePanel.this.page.getFeedbackPanel());
+                    return;
+                }
+                // generate inputs from file
+                File newFile;
+                Map<Object, Object> valueMap;
+                try {
+                    newFile = uploadedFile.writeToTempFile();
+                    String fileContent = Files.readFirstLine(newFile, Charset.defaultCharset());
+                    valueMap = JsonImporter.generateValuesFromString(fileContent);
+                    if (valueMap == null) {
+                        GeneratePanel.this.page.getFeedbackPanel().error("Values could not be read.");
+                        target.add(GeneratePanel.this.page.getFeedbackPanel());
+                        return;
+                    }
+                } catch (Exception e) {
+                    logger.warn("File could not be read", e);
+                    GeneratePanel.this.page.getFeedbackPanel().error("File could not be read.");
+                    target.add(GeneratePanel.this.page.getFeedbackPanel());
+                    return;
+                }
+
+                EapEventType newEventType = (EapEventType) valueMap.get("eventType");
+                if(selectedEventType.getID() != newEventType.getID()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') 'if' is not followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=checkstyle%3AWhitespace_Around_BP2016)